src/pages/MyLeaseDetails.js

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { getSingleLeaseAction } from '../redux/lease/lease';
+import { login } from '../redux/user/user';
+
+const MyLeaseDetails = () => {
+  const dispatch = useDispatch();
+  const params = useParams();
+  const { leaseDetails, user } = useSelector((state) => state);
+  const { lease, loading, err: error } = leaseDetails;
+
+  useEffect(() => {
+    if (user.userId) {
+      dispatch(getSingleLeaseAction(params.lease_id, user.userId));
+    }
+  }, [user]);
+
+  useEffect(() => {
+    if (localStorage.getItem('someRandomVitalData')) {
+      const { timestamp, mainUser } = JSON.parse(localStorage.getItem('someRandomVitalData'));
+      const now = new Date().getTime();
+      const oneDayInMillSecs = 86400000;
+      if (now - timestamp < (7 * oneDayInMillSecs)) {
+        dispatch(login(mainUser));
+      } else {
+        localStorage.removeItem('someRandomVitalData');
+      }
+    }
+  }, []);
```

Comment:
- [ ] Since this function is used in a lot of places, maybe it would be useful to put it in another file and call it when you need it.

Reply:
Thank you, @MuriloRoque :pray:
***Resolved*** :heavy_check_mark:

Reply:
![image](https://user-images.githubusercontent.com/45668340/157695382-a141007f-395f-42eb-82e8-2a8e740c8262.png)